tooling/operation/get-parameters-as-json-schema.js

```diff
@@ -15,173 +15,347 @@ const types = {
-            // Only add a type if the previous prop isn't also named `properties`!
+    if (example !== undefined) {
+      break;
+    }
+  }
+
+  return example;
+}
+
+/**
+ * Given an OpenAPI-flavored JSON Schema, make an effort to modify it so it's shaped more towards stock JSON Schema.
+ *
+ * Why do this?
+ *
+ *  1. OpenAPI 3.0.x supports its own flavor of JSON Schema that isn't fully compatible with most JSON Schema tooling
+ *    (like `@readme/oas-form` or `@rjsf/core`).
+ *  2. While validating an OpenAPI definition will prevent corrupted or improper schemas from occuring, we have a lot of
+ *    legacy schemas in ReadMe that were ingested before we had proper validation in place, and as a result have some
+ *    API definitions that will not pass validation right now. In addition to reshaping OAS-JSON Schema into JSON Schema
+ *    this library will also fix these improper schemas: things like `type: object` having `items` instead of
+ *    `properties`, `type: array` missing `items`, or `type` missing completely on a schema.
+ *  3. Additionally due to OpenAPI 3.0.x not supporting JSON Schema, in order to support the `example` keyword that OAS
+ *    supports, we need to do some work in here to remap it into `examples`. However, since all we care about in respect
+ *    to examples for usage within `@readme/oas-form`, we're only retaining primitives. This *slightly* deviates from
+ *    JSON Schema in that JSON Schema allows for any schema to be an example, but since `@readme/oas-form` can only
+ *    actually **render** primitives, that's what we're retaining.
+ *  4. Though OpenAPI 3.1 does support full JSON Schema, this library should be able to handle it without any problems.
+ *
+ * @link https://json-schema.org/draft/2019-09/json-schema-validation.html
+ * @link https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.3.md
+ * @param {Object} data
+ * @param {Object[]} prevSchemas
+ * @param {string} currentLocation
+ */
+function constructSchema(data, prevSchemas = [], currentLocation = '') {
+  const schema = { ...data };
+
+  // If this schema contains a `$ref`, it's circular and we shouldn't try to resolve it. Just return and move along.
+  if (schema.$ref) {
+    return {
+      $ref: schema.$ref,
+    };
+  }
+
+  // If this schema is malformed for some reason, let's do our best to repair it.
+  if (!('type' in schema) && !isPolymorphicSchema(schema) && !isRequestBodySchema(schema)) {
+    if ('properties' in schema) {
+      schema.type = 'object';
+    } else if ('items' in schema) {
+      schema.type = 'array';
+    } else {
+      // If we're processing a schema that has no types, no refs, and is just a lone schema, we should treat it at the
+      // bare minimum as a simple string so we make an attempt to generate valid JSON Schema.
+      schema.type = 'string';
+    }
+  }
+
+  // JSON Schema doesn't support OpenAPI-style examples so we need to reshape them a bit.
```

Comment:
I like this approach a lot, preferring json schema as the format of truth (which is the direction oas is going), and reworking oas examples into that format.

Reply:
Yeahhhh previously this library would take the OpenAPI-style and completely rebuild it into JSON Schema which caused us to have to add support for things we didn't account for. I much prefer to just assume it's okay and then add/remove whatever we need from there for stuff we can't handle or support.

Reply:
I found a bug in particular while rewriting this where we'd completely toss `properties` from an object under some specific circumstances. 😬